projects/wave-core/src/lib/operators/types/gdal-source-type.model.ts

```diff
@@ -143,10 +153,15 @@ export class GdalSourceType extends OperatorType {
                     return {
                         channel: channelDict.channel,
                         datatype: channelDict.datatype,
-                        file_name: channelDict.file_name,
+                        file_name: channelDict.file_name ? channelDict.file_name : undefined,
+                        path: channelDict.path ? channelDict.path : undefined,
                         unit: channelDict.unit.toDict(),
+                        netcdf_subdataset: channelDict.netcdf_subdataset ? channelDict.netcdf_subdataset : undefined,
                     };
                 }) : undefined,
+                file_name: this.gdalParams.file_name ? this.gdalParams.file_name : undefined,
+                path: this.gdalParams.path ? this.gdalParams.path : undefined,
+                netcdf_subdataset: this.gdalParams.netcdf_subdataset ? this.gdalParams.netcdf_subdataset : undefined,
```

Comment:
same here

Reply:
variable is not required by mapping here